pkg/service/K8sService.go

```diff
@@ -34,18 +36,87 @@ type K8sService interface {
 	PatchResource(ctx context.Context, restConfig *rest.Config, r *bean.KubernetesResourcePatchRequest) error
 }
 
+type HelmReleaseConfig struct {
+	EnableHelmReleaseCache    bool   `env:"ENABLE_HELM_RELEASE_CACHE" envDefault:"true"`
+	MaxCountForHelmRelease    int    `env:"MAX_COUNT_FOR_HELM_RELEASE" envDefault:"20"`
+	ManifestFetchBatchSize    int    `env:"MANIFEST_FETCH_BATCH_SIZE" envDefault:"2"`
+	RunHelmInstallInAsyncMode bool   `env:"RUN_HELM_INSTALL_IN_ASYNC_MODE" envDefault:"false"`
+	ParentChildGvkMapping     string `env:"PARENT_CHILD_GVK_MAPPING" envDefault:""`
+}
+
+func GetHelmReleaseConfig() (*HelmReleaseConfig, error) {
+	cfg := &HelmReleaseConfig{}
+	err := env.Parse(cfg)
+	return cfg, err
+}
+
 type K8sServiceImpl struct {
-	logger *zap.SugaredLogger
+	logger                *zap.SugaredLogger
+	helmReleaseConfig     *HelmReleaseConfig
+	gvkVsChildGvrAndScope map[schema.GroupVersionKind][]k8sCommonBean.GvrAndScope
+}
+
+func NewK8sServiceImpl(logger *zap.SugaredLogger, helmReleaseConfig *HelmReleaseConfig) *K8sServiceImpl {
+
+	gvkVsChildGvrAndScope := make(map[schema.GroupVersionKind][]k8sCommonBean.GvrAndScope)
+	k8sServiceImpl := &K8sServiceImpl{
+		logger:                logger,
+		helmReleaseConfig:     helmReleaseConfig,
+		gvkVsChildGvrAndScope: gvkVsChildGvrAndScope,
+	}
+	if len(helmReleaseConfig.ParentChildGvkMapping) > 0 {
+		err := k8sServiceImpl.populateParentChildGvkMapping(gvkVsChildGvrAndScope)
+		if err != nil {
+			return nil
+		}
+	}
+	return k8sServiceImpl
 }
 
-func NewK8sServiceImpl(logger *zap.SugaredLogger) *K8sServiceImpl {
-	return &K8sServiceImpl{
-		logger: logger,
+func (impl K8sServiceImpl) populateParentChildGvkMapping(gvkVsChildGvrAndScope map[schema.GroupVersionKind][]k8sCommonBean.GvrAndScope) error {
+	var gvkChildMappings []ParentChildGvkMapping
+	parentChildGvkMapping := impl.helmReleaseConfig.ParentChildGvkMapping
+	err := json.Unmarshal([]byte(parentChildGvkMapping), &gvkChildMappings)
+	if err != nil {
+		impl.logger.Errorw("error in unmarshalling ParentChildGvkMapping", "parentChildGvkMapping", parentChildGvkMapping, "err", err)
+		return err
+	}
+	for _, parent := range gvkChildMappings {
+		childGvrAndScopes := make([]k8sCommonBean.GvrAndScope, 0)
```

Comment:
1. make fix length array
2. use pointers
3. return value instead of caching directly

Reply:
done